### PR TITLE
App made fully customisable through styles.xml

### PR DESCRIPTION
**Pull request by mananwason**
@creativepsyco I think I have made everything customisable through style sheet. Please review and tell me if I have missed out anything. Thanks.

### CODE REVIEW

**Comment by mananwason** [comment]
@creativepsyco please review and merge.

**Comment by creativepsyco** [comment]
You have a separate commit for some other issue as well, can u separate it and file a PR for it?

**Comment by mananwason** [comment]
Sure.

**Comment by mananwason** [comment]
@creativepsyco Done. Please review now.

File: app/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -6,15 +6,31 @@
...
+        <item name="android:textColorSecondary">@color/color_text_dark</item>
     </style>
 
     <style name="DrawerArrowStyle" parent="Widget.AppCompat.DrawerArrowToggle">
         <item name="spinBars">true</item>
-        <item name="color">@android:color/white</item>
+        <item name="color">@color/color_accent</item>
+    </style>
+
+    <style name="FrameLayout">
```
Comment (creativepsyco) [inline app/src/main/res/values/styles.xml]: why the style name is `FrameLayout` choose something more logical.

File: app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -23,6 +39,13 @@
         <item name="android:background">#e5e5e5</item>
     </style>
 
+    <style name="SeparatorLineBlack" parent="SeparatorLine">
```
Comment (creativepsyco) [inline app/src/main/res/values/styles.xml]: you can use `SeparatorLine.Black` then you don't need to write the `parent` attribute.

File: app/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -33,20 +56,66 @@
         <item name="android:background">?attr/selectableItemBackground</item>
     </style>
 
+    <style name="Custom.Text.HeadLine" parent="TextAppearance.AppCompat.Headline">
```
Comment (creativepsyco) [inline app/src/main/res/values/styles.xml]: `Custom.Text` -> do you have `Custom` and `Custom.Text` defined somewhere?

File: app/src/main/res/values/styles.xml [hunk 4]
```diff
@@ -33,20 +56,66 @@
...
     <style name="Header">
         <item name="android:layout_width">match_parent</item>
         <item name="android:layout_height">wrap_content</item>
         <item name="android:paddingBottom">8dp</item>
         <item name="android:textSize">20sp</item>
-        <item name="android:textColor">@color/color_primary</item>
+        <item name="android:textColor">@color/color_text</item>
+    </style>
+
+    <style name="MyActionBarTitleText" parent="TextAppearance.AppCompat.Widget.ActionBar.Title">
```
Comment (creativepsyco) [inline app/src/main/res/values/styles.xml]: A better name please.

File: app/src/main/res/values/styles.xml [hunk 5]
```diff
@@ -33,20 +56,66 @@
...
+        <item name="android:textColor">@color/color_text</item>
+        <item name="android:textStyle">bold</item>
     </style>
 
-    <style name="Widget.CardContent" parent="android:Widget">
+    <style name="Widget.CardContent" parent="CardView">
```
Comment (creativepsyco) [inline app/src/main/res/values/styles.xml]: What is `CardView` is it defined somewhere?

File: app/src/main/res/values/styles.xml [hunk 6]
```diff
@@ -33,20 +56,66 @@
...
+        <item name="android:textColor">@color/color_text_dark</item>
         <item name="android:paddingLeft">16dp</item>
         <item name="android:paddingRight">16dp</item>
         <item name="android:paddingTop">24dp</item>
         <item name="android:paddingBottom">24dp</item>
         <item name="android:orientation">vertical</item>
+
+    </style>
+
+    <style name="Widget.SpeakerCardContent" parent="CardView">
```
Comment (creativepsyco) [inline app/src/main/res/values/styles.xml]: Same here, where is `CardView ` defined?

File: app/src/main/res/values/styles.xml [hunk 7]
```diff
@@ -33,20 +56,66 @@
...
+        <item name="android:textColor">@color/color_text_dark</item>
     </style>
 
+    <style name="ToolbarTheme" parent="Widget.AppCompat.Toolbar">
+        <item name="android:background">@color/color_primary</item>
+        <item name="android:textColorPrimary">@color/color_text_dark</item>
+
+    </style>
+
+    <style name="Custom.TextAppearance.AppCompat.Medium" parent="TextAppearance.AppCompat.Medium">
```
Comment (creativepsyco) [inline app/src/main/res/values/styles.xml]: Do you have `Custom.TextAppearance` and `Custom` defined somewhere?

File: app/src/main/res/values/styles.xml [hunk 8]
```diff
@@ -33,20 +56,66 @@
...
+        <item name="android:textColor">@color/color_text_dark</item>
+    </style>
+
+    <style name="Custom.TextAppearance.AppCompat.Body1" parent="TextAppearance.AppCompat.Body1">
```
Comment (creativepsyco) [inline app/src/main/res/values/styles.xml]: Same here.

File: app/src/main/res/values/styles.xml [hunk 9]
```diff
@@ -33,20 +56,66 @@
...
+        <item name="android:textColor">@color/color_text_dark</item>
+    </style>
+
+    <!-- Change Session Description (SessionDetailActivity) style -->
+    <style name="Custom.Text.SessionDescription" parent="Metadata">
```
Comment (creativepsyco) [inline app/src/main/res/values/styles.xml]: `Custom` and `Custom.Text` defined somewhere?

**Comment by creativepsyco** [comment]
Hi @mananwason 

The styles don't follow the correct inheritance guidelines for Android. Please read the [Android Guidelines][1] on themes and styles, and how you should structure them, using proper scopes for inheritance purposes. 

Some quick guidelines:

* Dont use `.` without understanding what it does.
* Use the `parent` attribute when the child style name is completely different, e.g. `Custom.Text` should not need a `parent` it already has one, i.e. `Custom` is the parent here, and `Custom.Text` is the style you are defining.
* Use proper meaningful names, dont have stuff like `MyToolbarStyle`, makes it look it, its been copied from somewhere
* Don't use view class names as Style parents or style names.
* Also check that parent styles are already defined.

Fix it up and I will check again.

Cheers,
[1]: http://developer.android.com/guide/topics/ui/themes.html

**Comment by mananwason** [comment]
okay @creativepsyco I'll fix everything and get back to you

**Pull request closed without merging**